src/foundry/foundry.js/roll.d.ts

```diff
@@ -1,501 +1,515 @@
+     * @defaultValue `'templates/dice/tooltip.html'`
+     */
+    static TOOLTIP_TEMPLATE: string;
 
-  /**
-   * Return an Array of the individual DiceTerm instances contained within this Roll.
-   */
-  get dice(): DiceTerm[];
+    /**
+     * Prepare the data structure used for the Roll.
+     * This is factored out to allow for custom Roll classes to do special data preparation using provided input.
+     * @param data - Provided roll data
+     * @returns The prepared data object
+     */
+    protected _prepareData(data: D): D;
 
-  /**
-   * Return a standardized representation for the displayed formula associated with this Roll.
-   */
-  get formula(): string;
+    /* -------------------------------------------- */
+    /*  Roll Attributes                             */
 
-  /**
-   * The resulting arithmetic expression after rolls have been evaluated
-   */
-  get result(): string;
+    /* -------------------------------------------- */
 
-  /**
-   * Return the total result of the Roll expression if it has been evaluated.
-   */
-  get total(): number;
+    /**
+     * Return an Array of the individual DiceTerm instances contained within this Roll.
+     */
+    get dice(): DiceTerm[];
 
-  /**
-   * @deprecated since 0.8.1
-   */
-  get _rolled(): boolean;
+    /**
+     * Return a standardized representation for the displayed formula associated with this Roll.
+     */
+    get formula(): string;
 
-  /**
-   * A factory method which constructs a Roll instance using the default configured Roll class.
-   * @typeParam D - the type of data object against which to parse attributes within the formula
-   * @param formula - The formula used to create the Roll instance
-   * @param data - The data object which provides component data for the formula
-   * @param options - Additional options which modify or describe this Roll
-   * @returns The constructed Roll instance
-   */
-  static create<D extends Record<string, unknown> = {}>(
-    formula: string,
-    data?: D,
-    options?: Partial<Roll.Options>
-  ): Roll<D>;
+    /**
+     * The resulting arithmetic expression after rolls have been evaluated
+     */
+    get result(): string;
 
-  /**
-   * Transform an array of RollTerm objects into a cleaned string formula representation.
-   * @param terms - An array of terms to represent as a formula
-   * @returns The string representation of the formula
-   */
-  static getFormula(terms: RollTerm[]): string;
+    /**
+     * Return the total result of the Roll expression if it has been evaluated.
+     */
+    get total(): number | undefined;
 
-  /**
-   * A sandbox-safe evaluation function to execute user-input code with access to scoped Math methods.
-   * @param expression - The input string expression
-   * @returns The numeric evaluated result
-   */
-  static safeEval(expression: string): number | unknown;
+    /**
+     * @deprecated since 0.8.1
+     */
+    get _rolled(): boolean;
```

Comment:
this seems to be in the wrong spot, in foundry.js it comes at the very end of the `Roll` class.

In general, it looks like from here on, the order of foundry.js is not preserved. That makes it really hard to review for me. I will continue with the review after that has been addressed.

Reply:
Here is everything in a wrong spot because the whole file was messed up. I guess it will be the best to remove everything from dice and rewrite it from scratch.